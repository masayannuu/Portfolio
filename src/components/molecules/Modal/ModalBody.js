import React from 'react'
import PropTypes from 'prop-types'
import { Image } from 'reactbulma'

const ModalBody = ({ imageUrl, description, teamData }) => (
  <div className="content">
    <Image src={imageUrl} />
    <h3>説明</h3>
    <p>{description}</p>
    <h3>チームについて</h3>
    <dl>
      <dt>人数</dt>
      <dd>{teamData.number}人</dd>
      <dt>役割</dt>
      <dd>{teamData.roll}</dd>
      <dt>やったこと</dt>
      <dd>{teamData.work}</dd>
    </dl>
  </div>
)

ModalBody.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  teamData: PropTypes.shape({
    number: PropTypes.number,
    roll: PropTypes.string,
    work: PropTypes.string
  }).isRequired
}

export default ModalBody
