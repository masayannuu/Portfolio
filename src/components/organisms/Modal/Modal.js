import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Button, Delete } from 'reactbulma'

const Modal = ({
  children, handleCloseModal, showModal, title
}) => {
  if (!showModal) {
    return null
  }

  const StyledBack = styled.div`
    background-color: #a2c4c7;
  `

  const StyledHeader = styled.header`
    background-color: #f0f0f0;
  `

  return (
    <div className="modal is-active">
      <StyledBack className="modal-background" />
      <div className="modal-card">
        <StyledHeader className="modal-card-head">
          <p className="modal-card-title">{title}</p>
          <Delete onClick={handleCloseModal} />
        </StyledHeader>
        <section className="modal-card-body">
          <div className="content">{children}</div>
        </section>
        <footer className="modal-card-foot">
          <Button white onClick={handleCloseModal}>
            Back
          </Button>
        </footer>
      </div>
    </div>
  )
}

Modal.propTypes = {
  handleCloseModal: PropTypes.func.isRequired,
  showModal: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired
}

export default Modal
