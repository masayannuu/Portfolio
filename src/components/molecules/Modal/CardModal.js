import React from 'react'
import PropTypes from 'prop-types'

import Modal from './Modal'
import ModalLink from '../../atoms/Link/ModalLink'
import ModalBody from './ModalBody'

class CardModal extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      title: props.title,
      imageUrl: props.imageUrl,
      description: props.description,
      teamData: props.teamData,
      informationTags: props.informationTags
    }
    this.handleOpenModal = this.handleOpenModal.bind(this)
    this.handleCloseModal = this.handleCloseModal.bind(this)
  }

  handleOpenModal () {
    this.setState({ showModal: true })
  }

  handleCloseModal () {
    this.setState({ showModal: false })
  }

  render () {
    const isTag = typeof this.state.informationTags !== 'undefined'
    return (
      <div className="tk-a-otf-gothic-bbb-pr6n">
        <ModalLink onClick={this.handleOpenModal} />
        <Modal
          handleCloseModal={this.handleCloseModal}
          showModal={this.state.showModal}
          title={this.state.title}
        >
          <ModalBody
            imageUrl={this.state.imageUrl}
            description={this.state.description}
            teamData={this.state.teamData}
            informationTags={this.state.informationTags}
            isTag={isTag}
          />
        </Modal>
      </div>
    )
  }
}

CardModal.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  teamData: PropTypes.shape({
    number: PropTypes.number,
    roll: PropTypes.string,
    work: PropTypes.string
  }).isRequired
}

export default CardModal
