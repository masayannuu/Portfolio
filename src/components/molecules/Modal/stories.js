import React from 'react'
import { storiesOf } from '@storybook/react'

import CardModal from './CardModal'
import Modal from './Modal'
import ModalBody from './ModalBody'

const description =
  'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字'

const team_data = {
  number: 3,
  roll: '設計・開発主担当',
  work:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ'
}

storiesOf('Molecules.CardModal', module).add('CardModal', () => (
  <CardModal
    title="File Transfer Application"
    imageUrl="https://bulma.io/images/placeholders/1280x960.png"
    description={description}
    teamData={team_data}
  />
))

storiesOf('Molecules.ModalBody', module).add('ModalBody', () => (
  <ModalBody
    title="File Transfer Application"
    imageUrl="https://bulma.io/images/placeholders/1280x960.png"
    description={description}
    teamData={team_data}
  />
))

storiesOf('Molecules.Modal', module).add('Modal', () => (
  <Modal
    handleCloseModal={() => console.log('handle close')}
    showModal
    title="File Transfer Application"
  >
    <p>{description}</p>
  </Modal>
))
