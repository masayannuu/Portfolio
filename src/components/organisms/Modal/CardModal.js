import React from 'react'

import Modal from './Modal'
import ModalLink from '../../atoms/Link/ModalLink'

class CardModal extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      title: props.title
    }
    this.handleOpenModal = this.handleOpenModal.bind(this)
    this.handleCloseModal = this.handleCloseModal.bind(this)
  }

  handleOpenModal () {
    this.setState({ showModal: true })
  }

  handleCloseModal () {
    this.setState({ showModal: false })
  }

  render () {
    return (
      <div>
        <ModalLink onClick={this.handleOpenModal} />
        <Modal
          handleCloseModal={this.handleCloseModal}
          showModal={this.state.showModal}
          title={this.state.title}
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet justo in arcu
            efficitur malesuada nec ut diam. Aenean a iaculis eros. Proin nec purus congue, rutrum
            sapien id, sodales ante. Nam imperdiet sapien pretium leo dapibus euismod. Ut ac
            venenatis nunc. Praesent viverra purus vel lacus ullamcorper porta a a augue. Proin
            rhoncus tempus leo sed ultricies. In luctus aliquam placerat. Cras efficitur enim vitae
            vulputate consequat. Nulla tellus est, fringilla quis nisi eu, aliquam finibus eros.
          </p>
        </Modal>
      </div>
    )
  }
}

export default CardModal
