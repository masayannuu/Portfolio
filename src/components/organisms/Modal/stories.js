import React from 'react'
import { storiesOf } from '@storybook/react'
import CardModal from './CardModal'

storiesOf('Organisms.CardModal', module).add('CardModal', () => <CardModal title="story" />)
