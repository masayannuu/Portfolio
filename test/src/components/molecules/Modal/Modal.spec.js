import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Modal from '../../../../../src/components/molecules/Modal/Modal'

const description =
  'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字'

describe('Modal Modal Molecules', () => {
  it('should render Modal', () => {
    const modal = (
      <Modal
        handleCloseModal={() => console.log('handle close')}
        showModal
        title="File Transfer Application"
      >
        <p>{description}</p>
      </Modal>
    )
    const wrapper = Enzyme.shallow(modal)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
