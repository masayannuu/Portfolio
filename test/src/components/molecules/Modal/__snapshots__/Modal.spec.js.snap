// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Modal Molecules should render Modal 1`] = `
<div
  className="modal is-active animated bounceInUp"
>
  <Modal__StyledBack
    className="modal-background"
  />
  <div
    className="modal-card"
  >
    <Modal__StyledHeader
      className="modal-card-head"
    >
      <p
        className="modal-card-title"
      >
        File Transfer Application
      </p>
      <Delete
        as="div"
        onClick={[Function]}
      />
    </Modal__StyledHeader>
    <section
      className="modal-card-body"
    >
      <div
        className="content"
      >
        <p>
          この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字
        </p>
      </div>
    </section>
    <Modal__StyledFooter
      className="modal-card-foot"
    >
      <Button
        as="button"
        className="animated zoomInUp"
        onClick={[Function]}
        white={true}
      >
        Back
      </Button>
    </Modal__StyledFooter>
  </div>
</div>
`;
